Fix for duplication of product which has both combinations and specific prices

matrixino opened the pull request:
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7
| Description?  | Fix for PSCSX-9312 & PSCSX-8996 - Error duplicating products with combinations and specific prices
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? |  PSCSX-9312, PSCSX-8996
| How to test?  | Duplicate a product which has both combinations and specific prices

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/8518)
<!-- Reviewable:end -->

matrixino committed:
Fix for PSCSX-9312 & PSCSX-8996

LittleBigDev commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8518#-:-KzY5JUNqGsD6OOmzwEn:bnfp4nl)*
<!-- Sent from Reviewable.io -->

LittleBigDev reviewed (dismissed):
Thank you @matrixino !

matrixino committed:
Update db_structure.sql

Fix slow query search when attributes are involved

matrixino committed:
Update (#1)

* Build assets

* BO: open edit configuration in a new page

* BO: Use multilanguage on delivery product shipping

* Add cover to allowed properties in ajax requests

Cover attribute is widely used by theme developers for ajax search

* CO: Merge module logo checks

* CO: Add path attribute in module data

* CO: Use ArrayCache for loaded modules

* CO: Add PrestaTrust feature as SF event

* BO: Prepare PrestaTrust data to display

* CO: Fix placeholder on translated string

* BO: Display PrestaTrust info in templates

* CO: Introduce confirmation steps in back end

* BO: Remove ajax checks in module controller as they are mandatory

* BO: Handle generic param for module actions params

* BO: Link modals and confrmation system

* CO: Refacto of Addons API data provider

* CO: Change module structure regarding API content

* CO: Add pico in modules, in list view

* BO: Reuse form of module card to trigger installation

* CO: Add API call for PrestaTrust check

* CO: Enable PrestaTrust checks only for concerned modules

* CO: Fix typo in module attributes loaded

* CO: Code review changes

* CO: Set module path a construct param of ModuleRepository

* CO: Changes following API checks

* CO: Fix integrity check for PSTrust

* CO: Wording of PrestaTrust msgs

* CO: Code review regarding PrestaTrust

* TE: Fixing module related tests

* CO: Code review regarding PrestaTrust v2

* TE: Add test for PrestaTrust

* TE: Mock modules folder

* CO: Add PHP doc

* CO: Code review

* BO: replace const with var

* TE: Remove now duplicated mock

* CO: Fix undefined array key after code review

* CO: Move module checks in zip management

* CO: Introduce ModuleZip and use it with related event

* TE: Update tests regarding PrestaTrust

* CO: Set default response when API unavailable

* CO: Check attribute value instead of its existence

* CO: Remove duplicate call to trans for modules

* BO: Handle PrestaTrust step on upload as well

* FO: Move linklist css outside .footer_container

* Improve cache :
- Use multiDelete to invalidate the cache in memcached
- Implement a lazy LRU to remove from the cache the less used objects
- Implement in memcached a mecanism which automatically detect when an object is too big
- Split the big "table => queries" cache into several caches

* FO: Generated assets

* improve coding style

* fix wrong cache behaviour

* add test

* add tests for cache

* php 5.4 compat

* avoid test failure if the class is already loaded

* fix php 5.4 failure

* fix php 5.4 failure

* properly invalidate all the related entries for queries which references several tables
add tests

* cs fixer on test

* post review fixes

* post review fixes

* Load UI kit on scss instead of js so that its classes can be extended

* Minify imported css

* Fix spacing on bulk actions from on the product page's combination section

* Fix category tags on product page

* Fix category search missing loupe on product page

* Fix horizontal scrolling on the product page when the screen size is < 1450px

* Fix date time picker not working on product page when the locale is different from english

* Upgrade momentjs library

* Fix add feature spacing on the product page

* Fix header search form misalignment

* Fix style of the header employee popup

* Fixed style of attachments on the product page

* Fix "No attachment yet" message even when there are attachments on the product page

* Fix related products style

* Add options to webpack config for compiling assets with source maps (commented out)

* Catalog controler wrongly use by default desc sorting instead of "last"

* Fix combination label cross dropping down to the next line on the product page on Firefox

* Build assets

* CO: improve order presenter performance

* CO: Remove risky and useless sprintf() in controllers

* CO: Remove risky and useless sprintf() in classes

* CO: Remove risky and useless sprintf() in CartPresenter

* CO: Remove sprintf for trans() with html code

* CO: Code review changes

* CO: fix notice error on maintenance page

* BO: Fix JS error on Performance page

* BO: Remove required status of cache servers

* extract amounts value for reuse

* post review fixes

* CO: force query build arg_separator to &

* CO: Shop domain is needed for several API call

Moved it as a global parameter of ApiClient

* BO: Display only module pictos in grid list

* BO: Set maximum display size for module icons

* Preventing any other data duplication

* Fixed `configuration_exists()` call

* Removed sprintf usage in SQL query

* [CO] Remove useless method from Customer class

* MO: access controller name in Modern pages

* fix wrong typehinting on attribute prices

* CO: Validate::isDateOrNull() should accept '0000-00-00'

* BO: fixed duplication of hooks in Product listing

* BO: removed all duplicated hooks except displayBackOfficeFooter

* TE: Add tests for Validate::isDateOrNull()

* CO: allow to sort bestsellers by sales

* CO: be able to identify a modern page in modules

* BO: customized Debug toolbar as PrestaShop toolbar

* requested changes by @LittleBigDev

* CO: updated files for Symfony 3 migration

* CO: fixed deprecations for Symfony 3

* CO: Fixes for errors about product page

* CO: fixed unresolved call to Request

* TE: moved PrestaShopBundle tests outside of Bundle

* CO: fixed coding style of HookManager class

* BO: Removed old emailing category

* BO: mibrated WebDebugConfig toolbar for Symfony3

* BO: add hooks on product catalogue page

* CO: added better description of displayDashboardToolbarTopMenu

* BO: fixed PrestaShop docs link

* BO: update catalog for 1.7.3

* Fix search when using exact words (#8473)

* FO: Fix search when using exact words

* TE: Extract keyword generation for test

* CO: module name in order details data

* FO: allow developers to add items to breadcrumb from tpl

* BO: missing translation for liveedit

* IN: Display generic exception message on cli installation error

* CO: refactored removeJS in ControllerCore

* CO: refactored smartyOutputContent in ControllerCore

* CO: refactored getField and getValue in AbstractFormCore

* BO: don't display error if AdminShopGroup doesn't exist

* CO: fix robots.txt exclude order param

The new param for ordering in ps 1.7 is order, not anymore orderby.
Robots are actually indexing page with order in params, it could consume crawl budget 
Orderway is also useless for 1.7

* FO: Fix shopping cart when allow checkout with wrongly product

* FO: Generate assets

* FO: Translated string associated a placeholder

* BO: Better use of jQuery selectors

* CO: switched to long array creation syntax in ControllerCore

* FO: Fix undefined offset on new products

* Fixed admin categry tree radio buttons positionning

* Update robots.txt

add disallow order & rm disallow orderway

* BO: Improved how detect PaymentModule type

BO: Improved how detect PaymentModule type

* Fixed category tree arrows behavior
(click for expand / collapse)

* BO: Avoid successive DOM modification when possible

* BO: Avoid loading of the same CLDR json files 3 times in //

* BO: fixed PrestaShop version on Addons Store url iframe

* BO: removed None option in cache adapters

* Build assets

* BO: Fix undefined 'smtp'

* TE: Add forgotten test file for CartPresenter

* BO: Avoid null as caching sysyem in shop params

* CO: missing translation for 1.7.3 because of a bug in our extract string tool

* BO: Update the default product order by ID desc

* TE: Add missing elements id

* CO: Fix array declaration

* Fix menu collapse button alignment

* TE: Run web installation with functionnal tests

* CO: Fix array declaration

* Unavailable product message is now the same for any action in cart page

* Fixed a translated string in CartController

* Edited a translated string to avoid duplicates in Crowdin

* BO: Update style of product combination tokens

* BO: Fix modal display on product page

* BO: fixed forms deprecation (Symfony3) on performance page

* FO: allow developers to add items to breadcrumb from tpl

* BO: Removed old emailing category

* CO: module name in order details data

* CO: refactored removeJS in ControllerCore

* CO: refactored smartyOutputContent in ControllerCore

* CO: refactored getField and getValue in AbstractFormCore

* BO: don't display error if AdminShopGroup doesn't exist

* CO: fix robots.txt exclude order param

The new param for ordering in ps 1.7 is order, not anymore orderby.
Robots are actually indexing page with order in params, it could consume crawl budget 
Orderway is also useless for 1.7

* CO: switched to long array creation syntax in ControllerCore

* BO: Show installed module first

* CO: updated files for Symfony 3 migration

* CO: fixed deprecations for Symfony 3

* CO: Fixes for errors about product page

* CO: fixed unresolved call to Request

* Update robots.txt

add disallow order & rm disallow orderway

* TE: moved PrestaShopBundle tests outside of Bundle

* CO: fixed coding style of HookManager class

* BO: mibrated WebDebugConfig toolbar for Symfony3

* CO: Fix array declaration

* CO: Fix array declaration

* BO: fixed forms deprecation (Symfony3) on performance page

* TE: updated namespace for develop

matrixino committed:
Merge pull request #2 from PrestaShop/develop

Update base
